api/src/main/java/org/eclipse/microprofile/openapi/models/media/Schema.java

```diff
@@ -847,18 +893,23 @@ default Schema format(String format) {
     }
 
     /**
-     * Returns the nullable property from this Schema instance which indicates whether null is a valid value.
+     * Returns whether the type property allows the object to be {@code null}
      *
-     * @return the nullable property
-     **/
+     * @return whether null is allowed
+     * @deprecated use {@link #getType()} and check if the result contains {@link SchemaType#NULL}
+     */
+    @Deprecated(since = "4.0")
     Boolean getNullable();
 
     /**
-     * Sets the nullable property of this Schema instance. Specify true if this Schema will allow null values.
+     * Updates the type property to either permit or disallow {@code null}
      *
      * @param nullable
-     *            a boolean value indicating this Schema allows a null value.
+     *            a boolean value indicating whether this Schema allows a null value.
+     * @deprecated use {@link #setType(List)}, {@link #addType(SchemaType)}, or {@link #removeType(SchemaType)} to add
+     *             or remove {@link SchemaType#NULL}
      */
+    @Deprecated(since = "4.0")
     void setNullable(Boolean nullable);
```

Comment:
I'm not sure whether it's worth keeping these methods or the single argument `setType`.

When I was working on the smallrye implementation, I found that these definitions didn't really retain backwards compatibility because
- `nullable` had three states: `true`, `false` and `null` (not set) whereas the current definition only allows for two. 

  Various parts of the code were checking whether `nullable` had been set by an earlier stage of a process, so the difference between `false` and `null` was important, even though they have the same meaning in the output document.
- it's not clear what `setNullable` should do if `type` is unset.

  If we set `type` to `["null"]` that forbids all types other than `null`, whereas `nullable = true` does nothing if `type` is not set in the old version.
  If we do nothing, then calling `setNullable(true); setType(OBJECT);` leads to a schema where null is not permitted.

It is possible to ensure that `setNullable()` and `setType()` work as they did before, but you need a hidden flag for the value of `nullable` which isn't output and the definition of what these two methods do becomes more complicated so I'm not sure that it's worth it.

Reply:
Either these need to be removed, or we need TCK tests to assert the specified behavior.

Reply:
For now I've removed them. I'd be happy to put them back after further discussion on how they should work.